examples/quantumgames/example_nonlocal_game.py

```diff
@@ -0,0 +1,489 @@
+#
+#   The referee.
+#
+# Alice and Bob want to play in a cooperative fashion against the referee.
+#
+# Now that we have set the stage with respect to the actors and actresses we will
+# encounter in this game, let us see how the game is actually played.
+#
+# .. figure:: ../../figures/nonlocal_game.svg
+#   :alt: nonlocal game
+#   :align: center
+#
+#   A two-player nonlocal game.
+#
+# A nonlocal game unfolds in the following manner.
+#
+# 1. The referee randomly generates questions denoted as :math:`x` and :math:`y`.
+#    The referee sends the question :math:`x` to Alice and the question
+#    :math:`y` to Bob. The referee also keeps a copy of :math:`x` and
+#    :math:`y` for reference.
+#
+# 2. Alice and Bob each receive their respective questions. They are then each
+#    expected to respond to their questions with answers that we denote as
+#    :math:`a` and :math:`b`. Alice sends :math:`a` to the referee, and Bob
+#    sends :math:`b`.
+#
+# 3. When the referee receives :math:`a` and :math:`b` from Alice and Bob,
+#    the referee evaluates a particular function that is predicated on the
+#    questions :math:`x` and :math:`y` as well as the answers :math:`a` and
+#    :math:`b`. The outcome of this function is either :math:`0` or
+#    :math:`1`, where an outcome of :math:`0` indicates a loss for Alice and
+#    Bob and an outcome of :math:`1` indicates a win for Alice and Bob.
+#
+#
+# Alice and Bob's goal in the above game is to get the function in Step-3 to
+# output a :math:`1`, or equivalently, to indicate a winning outcome. This type
+# of game is referred to as a *nonlocal game*.
+#
+# Classical and Quantum Strategies
+# --------------------------------
+#
+# Now that we have the framework for a nonlocal game, we can consider the
+# player's *strategy*; how the players play the game given access to certain
+# resources. There are a number of strategies that the players can use, but for
+# simplicity, we will restrict our attention to two types of strategies.
+#
+# 1. *Classical strategies*: The players answer the questions in a deterministic
+#    manner.
+#
+# 2. *Quantum strategies*: The players make use of quantum resources in the form
+#    of a shared quantum state and respective sets of measurements.
+#
+#
+# Classical strategies
+# ^^^^^^^^^^^^^^^^^^^^
+#
+# A *classical strategy* for a nonlocal game is one where the players
+# deterministically produce an output for every possible combination of inputs
+# they may receive in the game. The *classical value* of a nonlocal game is the
+# maximum probability achieved by the players over all classical strategies. For
+# a nonlocal game, :math:`G`, we use :math:`\omega(G)` to represent the classical
+# value of :math:`G`.
+#
+# One question you may have is whether a classical strategy can be improved by
+# introducing randomness. If the players randomly select their answers, is it
+
+# possible for them to do potentially better than if they had just played
```

Comment:
a blank line without a `#` breaks the formatting. 

https://toqito--1252.org.readthedocs.build/en/1252/auto_examples/quantumgames/example_nonlocal_game.html#classical-strategies